Bottom: 48630ee8e0910ac7dad446e37ca4a1579f2c9090
Top:    e48278013ab17c04429e10aa510564d75480e6eb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-13 14:57:37 +0000

Try a new, different sized VBO buffer for each upload.


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 8e5cd8b..d47cb21 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -147,10 +147,6 @@ hidgl_init_triangle_array (triangle_buffer *buffer)
    */
   buffer->use_map = false;
 
-  /* If using VBOs (but not mapping), we only need to this once */
-  if (buffer->use_vbo && !buffer->use_map)
-    glBufferData (GL_ARRAY_BUFFER, BUFFER_SIZE, NULL, GL_STREAM_DRAW);
-
   buffer->triangle_array = NULL;
   hidgl_reset_triangle_array (buffer);
 }
@@ -188,10 +184,10 @@ hidgl_flush_triangles (triangle_buffer *buffer)
       glUnmapBuffer (GL_ARRAY_BUFFER);
       buffer->triangle_array = NULL;
     } else {
-      /* NB: We only upload the portion of the buffer we've used */
-      glBufferSubData (GL_ARRAY_BUFFER, 0,
-                       BUFFER_STRIDE * buffer->vertex_count,
-                       buffer->triangle_array);
+      glBufferData (GL_ARRAY_BUFFER,
+                    BUFFER_STRIDE * buffer->vertex_count,
+                    buffer->triangle_array,
+                    GL_STREAM_DRAW);
     }
   } else {
     data_pointer = buffer->triangle_array;
